source/common/tracing/zipkin/zipkin_core_types.cc

```diff
@@ -220,7 +220,7 @@ const std::string Span::toJson() {
 void Span::finish() {
   // Assumption: Span will have only one annotation when this method is called
   SpanContext context(*this);
-  if (context.annotationSet().sr_ && !context.annotationSet().ss_) {
+  if (annotations_[0].value() == ZipkinCoreConstants::get().SERVER_RECV) {
```

Comment:
is it guaranteed that annotations_ has at least one element?

Reply:
The existing code accesses the first element when setting the endpoint - and the comment at the top of the method suggests that a single annotation is expected.

Reply:
Indeed, that is safe.